src/main/java/org/team1540/bunnybotTank2023/Robot.java

```diff
@@ -5,25 +5,71 @@
 
 package org.team1540.bunnybotTank2023;
 
-import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import org.littletonrobotics.junction.LogFileUtil;
+import org.littletonrobotics.junction.LoggedRobot;
+import org.littletonrobotics.junction.Logger;
+import org.littletonrobotics.junction.networktables.NT4Publisher;
+import org.littletonrobotics.junction.wpilog.WPILOGReader;
+import org.littletonrobotics.junction.wpilog.WPILOGWriter;
 
 /**
  * The VM is configured to automatically run this class, and to call the methods corresponding to
  * each mode, as described in the TimedRobot documentation. If you change the name of this class or
  * the package after creating this project, you must also update the build.gradle file in the
  * project.
  */
-public class Robot extends TimedRobot {
+public class Robot extends LoggedRobot {
     private Command autonomousCommand;
     
     private RobotContainer robotContainer;
 
     @Override
     public void robotInit() {
-        // Instantiate our RobotContainer.  This will perform all our button bindings, and put our
-        // autonomous chooser on the dashboard.
+        System.out.println("[Init] Starting AdvantageKit");
+        Logger logger = Logger.getInstance();
+
+        // Record metadata
+        logger.recordMetadata("ProjectName", BuildConstants.MAVEN_NAME);
+        logger.recordMetadata("BuildDate", BuildConstants.BUILD_DATE);
+        logger.recordMetadata("GitSHA", BuildConstants.GIT_SHA);
+        logger.recordMetadata("GitDate", BuildConstants.GIT_DATE);
+        logger.recordMetadata("GitBranch", BuildConstants.GIT_BRANCH);
+        switch (BuildConstants.DIRTY) {
+            case 0:
+                logger.recordMetadata("GitStatus", "All changes committed");
+                break;
+            case 1:
+                logger.recordMetadata("GitStatus", "Uncommitted changes");
+                break;
+            default:
+                logger.recordMetadata("GitStatus", "Unknown");
+        }
+
+        // Setup data receivers and replay source
+        switch(Constants.currentMode) {
+            case REAL:
+                if (DriverStation.isFMSAttached()) logger.addDataReceiver(new WPILOGWriter("/U"));
```

Comment:
why only when on real field?

Reply:
This is for logging to a USB, so unless we would plan on using the USB logs that come off of driver practice or testing, we shouldn't have uneccessary logging going on